core/src/OpenACC/Kokkos_OpenACC.cpp

```diff
@@ -59,11 +70,52 @@ void Kokkos::Experimental::OpenACC::impl_initialize(
         acc_get_device_num(acc_device_host);
   } else {
     using Kokkos::Impl::get_visible_devices;
+    acc_set_device_type(Impl::OpenACC_Traits::dev_type);
```

Comment:
This actually would break the falling back to host logic

Reply:
The falling back to host logic is handled in line 60, and thus this will not break it.
The `else` part in line 71 ~ 118 handles the case where device type is set to NVIDIA/AMD GPUs, host CPU, or unknown accelerator (`acc_device_not_host`).

Reply:
I am not familiar with the spec but I expected that if we call `acc_set_device_type(acc_device_not_host)` the runtime will error out if there is no device available for execution?
I that not how it works?  Is the runtime still allowed to fallback to host in that case?

Reply:
You are right, `acc_set_device_type(acc_device_not_host)` will issue an error if no device available for execution. 
In that sense, it will be better to use `acc_device_default` so that we can you whatever available in the target system. (changed the `OpenACC_Traits::dev_type` to `acc_device_default`)